application/actions/watches.go

```diff
@@ -113,3 +197,47 @@ func convertWatch(tx *pop.Connection, watch models.Watch, user models.User) (api
 
 	return output, nil
 }
+
+// convertWatchInput takes a `WatchInput` and either finds a record matching the UUID given in `input.ID` or
+// creates a new `models.Watch` with a new UUID. In either case, all properties that are not `nil` are set to the value
+// provided in `input`
+func convertWatchInput(tx *pop.Connection, input api.WatchInput, user models.User) (models.Watch, error) {
+	watch := models.Watch{}
+
+	watch.OwnerID = user.ID
+	watch.Name = input.Name
+
+	watch.SearchText = models.ConvertStringPtrToNullsString(input.SearchText)
+
+	if input.Size == nil {
+		watch.Size = nil
+	} else {
+		apiSize := *input.Size
+		s := models.GetRequestSizeFromAPISize(apiSize)
+		watch.Size = &s
+	}
+
+	if input.MeetingID == nil || *input.MeetingID == "" {
+		watch.MeetingID = nulls.Int{}
+	} else {
+		var meeting models.Meeting
+		if err := meeting.FindByUUID(tx, *input.MeetingID); err != nil {
+			return watch, err
+		}
+		watch.MeetingID = nulls.NewInt(meeting.ID)
+	}
+
+	return watch, nil
+}
+
+func convertLocation(input api.LocationInput) models.Location {
```

Comment:
I suspect we may need a `convertLocation` for output types.
```suggestion
func convertLocationInput(input api.LocationInput) models.Location {
```

Reply:
:+1: